File: lib/models/acronym.dart
enum AcronymCategory {
  adjectif('Adjectif'),
  adverbe('Adverbe'),
  conjonction('Conjonction'),
  interjection('Interjection'),
  motInvariable('Mot invariable'),
  nomFeminin('Nom Féminin'),
  nomMasculin('Nom Masculin'),
  nomPropre('Nom Propre'),
  sensFigure('Sens Figuré'),
  nomMasculinetsensFigure('Nom Masculin / Sens Figuré'),
  nomFemininetsensFigure('Nom Féminin / Sens Figuré'),
  verbeTransitifetsensFigure('Verbe Transitif / Sens Figuré'),
  verbeIntransitifetsensFigure('Verbe Intransitif / Sens Figuré'),
  adjectifetsensFigure('Adjectif / Sens Figuré'),
  nomPropreetsensFigure('Nom Propre / Sens Figuré'),
  verbePronominaletsensFigure('Verbe Pronominal / Sens Figuré'),
  adjectifetparticipePresent('Adjectif / Participe Présent'),
  nomMasculinetadjectif('Nom Masculin / Adjectif'),
  verbeIntransitif('Verbe Intransitif'),
  verbePronominal('Verbe Pronominal'),
  verbeTransitif('Verbe Transitif'),
  pronomPersonnel('Pronom Personnel'),
  pronomRelatifOuIndefini('Pronom Relatif ou Indéfini');

  final String label;
  const AcronymCategory(this.label);

  static List<AcronymCategory> get all => AcronymCategory.values;
}

enum TriAcronyme {
  alphabetique,
  categorie,
  aucun,
}

class Acronym {
  final String id;
  final String letters;
  final String fullForm;
  final List<String> examples;
  final AcronymCategory category;

  Acronym({
    required this.id,
    required this.letters,
    required this.fullForm,
    required this.examples,
    required this.category,
  });

  Map<String, dynamic> toJson() => {
        'id': id,
        'letters': letters,
        'fullForm': fullForm,
        'examples': examples,
        'category': category.label,
      };

  factory Acronym.fromJson(Map<String, dynamic> json) => Acronym(
        id: json['id'] as String,
        letters: json['letters'] as String,
        fullForm: json['fullForm'] as String,
        examples: (json['examples'] as List).map((e) => e.toString()).toList(),
        category: AcronymCategory.values.firstWhere(
          (c) => c.label == json['category'],
          orElse: () => AcronymCategory.nomMasculin,
        ),
      );
}

File: Le_DARIEL/Models/Acronym.swift
//
//  Acronym.swift
//  Le_DARIEL
//
//  Created by FAEI971 on 23/07/2025.
//

import Foundation

struct Acronym: Identifiable, Codable {
    let id: UUID
    var letters: String
    var fullForm: String
    var examples: [String]
    var category: Category
    
    enum Category: String, Codable, CaseIterable {
        case adjectif = "Adjectif"
        case adverbe = "Adverbe"
        case conjonction = "Conjonction"
        case interjection = "Interjection"
        case motInvariable = "Mot invariable"
        case nomFeminin = "Nom Féminin"
        case nomMasculin = "Nom Masculin"
        case nomPropre = "Nom Propre"
        case sensFigure = "Sens Figuré"
        case nomMasculinetsensFigure = "Nom Masculin / Sens Figuré"
        case nomFemininetsensFigure = "Nom Féminin / Sens Figuré"
        case verbeTransitifetsensFigure = "Verbe Transitif / Sens Figuré"
        case verbeIntransitifetsensFigure = "Verbe Intransitif / Sens Figuré"
        case adjectifetsensFigure = "Adjectif / Sens Figuré"
        case nomPropreetsensFigure = "Nom Propre / Sens Figuré"
        case verbePronominaletsensFigure = "Verbe Pronominal / Sens Figuré"
        case AdjectifetparticipePresent = "Adjectif / Participe Présent"
        case nomMasculinetadjectif = "Nom Masculin / Adjectif"
        case verbeIntransitif = "Verbe Intransitif"
        case verbePronominal = "Verbe Pronominal"
        case verbeTransitif = "Verbe Transitif"
        case pronomPersonnel = "Pronom Personnel"
        case pronomRelatifOuIndéfini = "Pronom Relatif ou Indéfini"
    }
    
    init(id: UUID = UUID(), letters: String, fullForm: String, examples: [String], category: Category = .nomMasculin) {
        self.id = id
        self.letters = letters
        self.fullForm = fullForm
        self.examples = examples
        self.category = category
    }
}

enum TriAcronyme {
    case alphabetique
    case categorie
    case aucun
}





